docs/source/specifications/projjson.rst

```diff
@@ -4,35 +4,83 @@
+
+Normative references
+--------------------
+
+The PROJJSON specification requires prior knowledge of the following normative
+specifications:
 
-PROJJSON is available as input and output of PROJ since PROJ 6.2.
+- `The JavaScript Object Notation (JSON) Data Interchange Format / IETF RFC 7159 <https://datatracker.ietf.org/doc/html/rfc7159>`_
+- `WKT2:2019 / ISO-19162:2019: Geographic information - Well-known text representation of coordinate reference systems  <http://docs.opengeospatial.org/is/18-010r7/18-010r7.html>`_
 
-The current version is 0.4.
+Definitions
+-----------
+
+- JavaScript Object Notation (JSON), and the terms object, member,
+  name, value, array, number, true, false, and null, are to be
+  interpreted as defined in [RFC7159].
+
+- integer: JSON number whose value has no fractional/exponent part.
+
+- All `term and definitions <http://docs.opengeospatial.org/is/18-010r7/18-010r7.html#4>`_
+  from WKT2:2019 apply.
 
 Schema
 ------
 
-A JSON schema of its grammar is available at
+A JSON schema of PROJJSON grammar is available at
 https://proj.org/schemas/v0.4/projjson.schema.json
 
-History
--------
+This schema defines a minimum set of constraints that apply to well-formed PROJJSON.
+Number of specific CRS and coordinate operation domain constraints are not expressed
+as JSON schema constraints: unless otherwise stated, the constraints (optional/mandatory/conditional
+character of information, restricted set of allowed values, etc.) defined
+in the WKT2:2019 specification also apply, as supplement to the JSON schema constraints.
+
+History of the schema
+---------------------
 
 * v0.4: additional properties allowed in id object (version, authority_citation, uri)
 * v0.3: additional properties allowed in BoundCRS object (name, scope, area, bbox, usages, remarks, id, ids)
 * v0.2: addition of geoid_model in VerticalCRS object.
 * v0.1: initial version for PROJ 6.2
 
-Content
--------
+Specification 
+-------------
 
-The high level objects are:
+A PROJJSON text is a JSON object which has, at a minimum, a required ``type`` member,
+whose value is a string describing the nature of the encoded geodetic object.
+
+An optional ``$schema`` member may be present, with its value being a string with
+a URL that points to the JSON schema that applies.
+
+Objects may be composed of sub-objects (e.g a GeographicCRS is made of a Datum or DatumEnsemble
+and a coordinate system). The ``type`` member of the sub-objects can be omitted when
+there is no ambiguity. For example, in the object which is the value of a ``coordinate_system``
+member, the ``type`` may be omitted. But for the value of the ``datum`` object of
+a GeographicCRS the ``type`` should be specified, as it can be either a GeodeticReferenceFrame
+or a DynamicGeodeticReferenceFrame.
+More formally, if the JSON schema specifies alternative types for the value of a
```

Comment:
Not quite clear.  Perhaps:

```
The ``type`` should also be specified if the JSON schema specifies alternative types for the value of a member using the oneOf constrct and those alternative types have a ``type`` member.  Otherwise it may be omitted.

```

Reply:
done